olsync/olsync.py

```diff
@@ -113,13 +120,25 @@ def sync_func(files_from, create_file_at_to, from_exists_in_to, from_equal_to_to
                         'Warning %s file will be overwritten by %s. Continue?' % (to_name, from_name)):
                     continue
 
-                click.echo("%s syncing from %s to %s." % (name, from_name, to_name))
+                click.echo("%s syncing from %s to %s." %
+                           (name, from_name, to_name))
                 create_file_at_to(name)
             else:
-                click.echo("%s file is equal to %s file. No sync necessary." % (name, to_name))
+                click.echo(
+                    "%s file is equal to %s file. No sync necessary." % (name, to_name))
         else:
-            click.echo("%s does not exist on %s. Creating file." % (name, to_name))
-            create_file_at_to(name)
+            click.echo("%s does not exist on %s. Creating file (directory)." %
+                       (name, to_name))
+
+            # deal with folders, _dir == name if `name` is directory
+            _dir = os.path.dirname(name)
+            if _dir and (not os.path.exists(_dir)):
+                # non-empty _dir
+                os.makedirs(_dir)
+
+            # `name` itself is not dir
+            if not os.path.isdir(name):
```

Comment:
Maybe `os.path.isfile(name)` is a bit more verbose?

Reply:
Agreed. They basically do the same job. Will use `os.path.isfile()` instead.